release_notes.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python
+
+from jira import JIRA
+import itertools, sys
+
+if len(sys.argv) < 2:
+    print >>sys.stderr, "Usage: release_notes.py <version>"
+    sys.exit(1)
+
+version = sys.argv[1]
+minor_version_dotless = "".join(version.split(".")[:3]) # i.e., 0.10.0 if version == 0.10.0.1
+
+JIRA_BASE_URL = 'https://issues.apache.org/jira'
+MAX_RESULTS = 100 # This is constrained for cloud instances so we need to fix this value
+
+def get_issues(jira, query, **kwargs):
+    """
+    Get all issues matching the JQL query from the JIRA instance. This handles expanding paginated results for you. Any additional keyword arguments are forwarded to the JIRA.search_issues call.
+    """
+    results = []
+    startAt = 0
+    new_results = None
+    while new_results == None or len(new_results) == MAX_RESULTS:
+        new_results = jira.search_issues(query, startAt=startAt, maxResults=MAX_RESULTS, **kwargs)
+        results += new_results
+        startAt += len(new_results)
+    return results
+
+def issue_link(issue):
+    return "%s/browse/%s" % (JIRA_BASE_URL, issue.key)
+
+
+if __name__ == "__main__":
+    apache = JIRA(JIRA_BASE_URL)
+    issues = get_issues(apache, 'project=KAFKA and fixVersion=%s' % version)
+    if not issues:
+        print >>sys.stderr, "Didn't find any issues for the target fix version"
+        sys.exit(1)
+
+    unresolved_issues = [issue for issue in issues if issue.fields.resolution is None]
+    if unresolved_issues:
+        for issue in unresolved_issues:
+            print >>sys.stderr, "Unresolved issue: %s %s" % (issue.key, issue_link(issue))
+        sys.exit(1)
+
+    # Get list of (issue type, [issues]) sorted by the issue ID type's , with each subset of issues sorted by their key so they are in
+    # increasing order of bug #
```

Comment:
To be honest, I considered removing the categories altogether. I actually can't tell if the ordering from JIRA is meaningful. It doesn't correspond to alphabetical, issue type ID, or anything else I could figure out. And unfortunately since JIRA is super configurable, you can't rely on the exact set of issue types.

I've changed the sorting so we'll prioritize the two you mentioned and the rest go in the order of their issue type IDs. We can add further customization if we want to refine the ordering of other types too.